products/memgraph/upcoming/tutorials/replication.md

```diff
@@ -0,0 +1,136 @@
+binary. Look [here](https://docs.memgraph.com/memgraph/getting-started/installation/docker-installation) for instructions.
+In addition, make sure you have either [mg_client](https://github.com/memgraph/mgclient/tree/release/1.0) or [mgconsole](https://github.com/memgraph/mgconsole) installed.
+
+We have to setup the cluster nodes' ports properly, so we'll use the following
+mapping:
+* the main node will have port 7687 assigned
+* the sync node will have port 7688 assigned
+* the async node will have port 7689 assigned
+* the sync-with-timeout node will have port 7690 assigned .
+Note that we could've chosen arbitrary ports.
+
+We fire up the terminal, and for each Memgraph instance (node)  we have to start, we'll
+run:
+
+```plaintext
+docker run --rm -p PORT:7687 memgraph:1.3.0-enterprise
+```
+
+where PORT runs from 7687 to 7690. The above commands will start a Memgraph
+node, and map it to the appropriate port on your host machine.
+
+Now, to set up the cluster, we'll have to start an mg_client or mgconsole
+instance for every running Memgraph node, and connect to it. To do this, we
+have to figure out their IP addresses. Running
+
+```plaintext
+docker ps
+```
+
+will list all the running docker processes, along with their "CONTAINER ID" and
+"NAME". Running
+
+```plaintext
+docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' container_name_or_id
+```
+
+where "container_name_or_id" is either the "CONTAINER ID" or "NAME" listed by
+`docker ps`, we can get the IP addresses of all the nodes. Once we have them, we
+can start setting up the replication. Let's say the ip addresses are as follows:
+* main:                      172.17.0.2
+* sync replica:              172.17.0.3
+* async replica:             172.17.0.4
+* sync-with-timeout replica: 172.17.0.5 .
+
+Let's assume we're using mgconsole to connect to and query the nodes. Firstly,
+we have to set up the replicas. We connect to a replica by running
+
+```plaintext
+mgconsole --host REPLICA_IP_ADDRESS --port REPLICA_PORT --use-ssl=false
+```
+
+where REPLICA_IP_ADDRESS is the address we found in the previous step, and the
+REPLICA_PORT is the port corresponding to that address. Remember, nodes with
+ports 7688 to 7690 are replicas.
+
+Once we're connected to a replica, we set its replication role to "REPLICA" by
+issuing
+
+```plaintext
+SET REPLICATION ROLE TO REPLICA WITH PORT 10000
+```
+
+Note that the port 10000 is arbitrary, and any available port number may be used
+instead. We repeat the process for all replicas.
+
+Now, it's time to set up the main. Again, we connect to the main using
+mgconsole:
+
+```plaintext
+mgconsole --host 172.17.0.2 --port 7687 --use-ssl=false
+```
+
+Then, for every replica, we issue the query that registers it:
+
+```plaintext
+REGISTER REPLICA r1 SYNC TO "172.17.0.3:10000";
+REGISTER REPLICA r2 ASYNC TO "172.17.0.4:10000";
+REGISTER REPLICA r3 SYNC WITH TIMEOUT 0.001 TO "172.17.0.5:10000";
```

Comment:
`0.001` is 1ms? Maybe it's better to replace it with `1` of `0.1` that's a more realistic real-world scenario.

Reply:
Sure.